app/controllers/check_ins_controller.rb

```diff
@@ -42,14 +43,14 @@ def update
   end
 
   private def merge_worked_at_and_account_into_log_entries(check_in_params)
-    check_in_params[:log_entries_attributes].each do |_id, log_entry|
+    check_in_params.fetch(:log_entries_attributes, {}).each do |_id, log_entry|
```

Comment:
Here, we're going into the `check_in` and digging out the `log_entries_attributes`  and the hash that we get from the new `check_in`, and iterating over it to get the `id` and the `log_entry` itself